### PR TITLE
Doc updates

### LINKED ISSUE #7422: Document that python setup.py develop add the symlink to easy-install.pth
Else, it's really hard to "uninstall" matplotlib.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

- Document using pip for installation throughout (closes #7422).
- manylinux wheels are now available and a reasonable approach too.
- Move the git install faq below the non-git install faq.
- Most GUI toolkits are now pip and/or conda-installable on recent Pythons (GObject is the main outlier), update the venv docs as such.  In other cases, https://github.com/stuaxo/vext can be used to punch a hole from the venvs to system-site-packages in a controlled manner.
- Dedent the bullet list at http://matplotlib.org/devdocs/faq/troubleshooting_faq.html#getting-help, which currently appears within a blockquote.
- Suggest using `python -c "from pylab import *; plot(); show()" --verbose-helpful` for a minimal error message, which avoids needing to create a separate script file.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant 
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: doc/faq/installing_faq.rst [hunk 1]
```diff
@@ -267,14 +178,14 @@ retry the install.  If that does not work, then check
 Installing via OSX mpkg installer package
 -----------------------------------------
 
-matplotlib also has a disk image (``.dmg``) installer, which contains a
-typical Installer.app package to install matplotlib.  You should use binary
+Matplotlib also has a disk image (``.dmg``) installer, which contains a
```
Comment (tacaswell) [inline doc/faq/installing_faq.rst]: This is no longer true, we only do wheels on mac now.  Can you remove all references to `dmg` files?

File: doc/faq/troubleshooting_faq.rst [hunk 2]
```diff
@@ -149,24 +143,22 @@ tracker only periodically.  If your problem has been determined to be
...
 
-First make sure you have a clean build and install (see
-:ref:`clean-install`), get the latest git update, install it and run a
-simple test script in debug mode::
+First make sure you have a clean build and install (see :ref:`clean-install`),
+get the latest git update, install it and run a simple test script in debug
+mode::
 
-    rm -rf build
     rm -rf /path/to/site-packages/matplotlib*
```
Comment (tacaswell) [inline doc/faq/troubleshooting_faq.rst]: This can be replaced by `pip unistall matplotlib`

File: doc/faq/installing_faq.rst (hunk 1 above)
Comment (anntzer) [reply doc/faq/installing_faq.rst]: Can you just edit accordingly?  I have close to no idea how things work on OSX, so I can cargo it but I'd rather not.

File: doc/faq/troubleshooting_faq.rst (hunk 2 above)
Comment (anntzer) [reply doc/faq/troubleshooting_faq.rst]: I intentionally left it to cover the case of "seriously broken install" but not sure it's worth it -- let me know what you think.

File: doc/faq/installing_faq.rst (hunk 1 above)
Comment (tacaswell) [reply doc/faq/installing_faq.rst]: I _also_ have no idea how macs work, but I do know we do not build a dmg file.

Comment (anntzer) [reply doc/faq/installing_faq.rst]: just killed the section

File: doc/faq/troubleshooting_faq.rst (hunk 2 above)
Comment (tacaswell) [reply doc/faq/troubleshooting_faq.rst]: That is a compelling position, including a reference to `pip uninstall matplotlib` someplace would be good though.

File: doc/faq/installing_faq.rst [hunk 3]
```diff
@@ -340,14 +205,71 @@ or::
...
+
+      su -c 'yum-builddep python-matplotlib'
+
+   This does not build Matplotlib, but it does get all of the build
+   dependencies, which will make building from source easier.
+
+If you want to be able to follow the development branch as it changes
+just replace the last step with::
+
+   pip install -e .
```
Comment (jklymak) [inline doc/faq/installing_faq.rst]: One gotcha that I've run into is `pip` not being installed in the conda environment I'm in, so it uses the default `pip`, and installs the development version over the released version in my main conda environment.  Not sure where environments fit into this document, but something that could be discussed.

Comment (anntzer) [reply doc/faq/installing_faq.rst]: I updated the thing to use `python -mpip` throughout, which should make sure pip always works from (and thus, in) the local environment.
I also think that conda now installs pip by default in all envs?  (but it's even if it doesn't it's not a problem as long as you use `python -mpip`).

**Comment by tacaswell** [comment]
I am :+1: on what @anntzer wrote, but added a few commits so I don't want to merge this.  Anyone (including @anntzer ) should merge this if it is a net improvement to the docs.

File: doc/faq/installing_faq.rst [hunk 4]
```diff
@@ -42,127 +34,46 @@ If you are still having trouble, see :ref:`reporting-problems`.
...
 Linux Notes
 ===========
 
-Because most Linux distributions use some sort of package manager,
-we do not provide a pre-built binary for the Linux platform.
-Instead, we recommend that you use the "Add Software" method for
-your system to install matplotlib. This will guarantee that everything
-that is needed for matplotlib will be installed as well.
+To install Matplotlib at the system-level we recommend that you use your 
+distribution's package manager to install Matplotlib.  This will guarantee
```
Comment (jklymak) [inline doc/faq/installing_faq.rst]: Suggest drop: "...to install Matplotlib"

File: doc/faq/installing_faq.rst [hunk 5]
```diff
@@ -42,127 +34,46 @@ If you are still having trouble, see :ref:`reporting-problems`.
...
-that is needed for matplotlib will be installed as well.
+To install Matplotlib at the system-level we recommend that you use your 
+distribution's package manager to install Matplotlib.  This will guarantee 
+that Matplotlib's dependencies will be installed as well.
 
-If, for some reason, you can not use the package manager, Linux usually
-comes with at least a basic build system. Follow the :ref:`instructions
-<install-from-git>` found above for how to build and install matplotlib.
+If, for some reason, you can not use the package manager, you may use the
+manylinux wheels available on PyPI::
```
Comment (jklymak) [inline doc/faq/installing_faq.rst]: OK, when I read this I thought that `manylinux` was a typo.  Put in quotes?

Comment (tacaswell) [reply doc/faq/installing_faq.rst]: https://github.com/pypa/manylinux They never use quotes, I do not think we want to deviate from their notation.

Comment (anntzer) [reply doc/faq/installing_faq.rst]: It's a bit jargon-esque but I'd prefer leaving it as it is.

Comment (jklymak) [reply doc/faq/installing_faq.rst]: Not a big deal, but how about "...you may use the wheels available from manylinux, available on PyPI"

Comment (anntzer) [reply doc/faq/installing_faq.rst]: `use the manylinux-tagged wheels, available on PyPI` OK?

Comment (jklymak) [reply doc/faq/installing_faq.rst]: Sounds good!

Comment (anntzer) [reply doc/faq/installing_faq.rst]: Upon further throught I just dropped the "manylinux", as they are the only kind of linux wheels on pypi anyways, and either the reader knows what it means in which case it's obvious the wheels should be manylinux-tagged, or the reader doesn't and that doesn't help him at all.

File: doc/faq/installing_faq.rst [hunk 6]
```diff
@@ -42,127 +34,46 @@ If you are still having trouble, see :ref:`reporting-problems`.
...
-your system to install matplotlib. This will guarantee that everything
-that is needed for matplotlib will be installed as well.
+To install Matplotlib at the system-level, we recommend that you use your
+distribution's package manager.  This will guarantee that Matplotlib's
+dependencies will be installed as well.
 
-If, for some reason, you can not use the package manager, Linux usually
-comes with at least a basic build system. Follow the :ref:`instructions
-<install-from-git>` found above for how to build and install matplotlib.
+If, for some reason, you can notuse the package manager, you may use the wheels
```
Comment (jklymak) [inline doc/faq/installing_faq.rst]: Notuse -> not use

Comment (anntzer) [reply doc/faq/installing_faq.rst]: removed the wrong space :) fixed

**Comment by tacaswell** [comment]
backported as 22c088e2f1e0ac1a302b5fddf6ce734c518e6bea

**Pull request merged**